src/OVAL/probes/unix/linux/rpminfo.c

```diff
@@ -303,18 +303,14 @@ static int get_rpminfo (struct rpminfo_req *req, struct rpminfo_rep **rep)
 
 void *probe_init (void)
 {
-	probe_offline_flags offline_mode = PROBE_OFFLINE_NONE;
+	probe_setoption(PROBEOPT_OFFLINE_MODE_SUPPORTED, PROBE_OFFLINE_CHROOT|PROBE_OFFLINE_RPMDB);
+	addMacro(NULL, "_dbpath", NULL, getenv("OSCAP_PROBE_RPMDB_PATH"), 0);
 
         if (rpmReadConfigFiles ((const char *)NULL, (const char *)NULL) != 0) {
                 dI("rpmReadConfigFiles failed: %u, %s.", errno, strerror (errno));
                 return (NULL);
         }
 
-        probe_getoption(PROBEOPT_OFFLINE_MODE_SUPPORTED, NULL, &offline_mode);
-        if (offline_mode & PROBE_OFFLINE_RPMDB) {
```

Comment:
But there is no check for the mode inside the probe and there was one before. I don't think it makes sense to use an implementation detail when we can use openscap API to get the mode...